From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <heftig@archlinux.org>
Date: Wed, 18 Mar 2026 12:11:17 +0100
Subject: [PATCH] Revert "Bug 2023494 [Wayland] Add support of wl_fixes
 r=emilio"

This reverts commit aa7c40968ad6a43c6569a5bb1e1add806ff6b292.

See https://bugzilla.mozilla.org/show_bug.cgi?id=2024157
---
 widget/gtk/mozwayland/mozwayland.c |   1 -
 widget/gtk/mozwayland/mozwayland.h | 113 -----------------------------
 2 files changed, 114 deletions(-)

diff --git a/widget/gtk/mozwayland/mozwayland.c b/widget/gtk/mozwayland/mozwayland.c
index 869afb1845b5..64d9843663ab 100644
--- a/widget/gtk/mozwayland/mozwayland.c
+++ b/widget/gtk/mozwayland/mozwayland.c
@@ -20,7 +20,6 @@ const struct wl_interface wl_callback_interface;
 const struct wl_interface wl_compositor_interface;
 const struct wl_interface wl_data_device_interface;
 const struct wl_interface wl_data_device_manager_interface;
-const struct wl_interface wl_fixes_interface;
 const struct wl_interface wl_keyboard_interface;
 const struct wl_interface wl_output_interface;
 const struct wl_interface wl_pointer_interface;
diff --git a/widget/gtk/mozwayland/mozwayland.h b/widget/gtk/mozwayland/mozwayland.h
index 08d13311cb10..17d4b8c86969 100644
--- a/widget/gtk/mozwayland/mozwayland.h
+++ b/widget/gtk/mozwayland/mozwayland.h
@@ -591,119 +591,6 @@ struct moz_wl_pointer_listener {
 #  define WL_POINTER_AXIS_VALUE120_SINCE_VERSION 8
 #endif
 
-#ifndef WL_FIXES_ERROR_ENUM
-#  define WL_FIXES_ERROR_ENUM
-/**
- * @ingroup iface_wl_fixes
- * wl_fixes error values
- *
- * These errors can be emitted in response to wl_fixes requests.
- */
-enum wl_fixes_error {
-  /**
-   * unknown global or the global is not removed
-   */
-  WL_FIXES_ERROR_INVALID_ACK_REMOVE = 0,
-};
-#endif /* WL_FIXES_ERROR_ENUM */
-
-#ifndef WL_FIXES_DESTROY
-#  define WL_FIXES_DESTROY 0
-#endif
-
-#ifndef WL_FIXES_DESTROY_REGISTRY
-#  define WL_FIXES_DESTROY_REGISTRY 1
-#endif
-
-#ifndef WL_FIXES_ACK_GLOBAL_REMOVE
-#  define WL_FIXES_ACK_GLOBAL_REMOVE 2
-#endif
-
-#ifndef WL_FIXES_DESTROY_SINCE_VERSION
-#  define WL_FIXES_DESTROY_SINCE_VERSION 1
-#endif
-
-#ifndef WL_FIXES_DESTROY_REGISTRY_SINCE_VERSION
-#  define WL_FIXES_DESTROY_REGISTRY_SINCE_VERSION 1
-#endif
-
-#ifndef WL_FIXES_ACK_GLOBAL_REMOVE_SINCE_VERSION
-#  define WL_FIXES_ACK_GLOBAL_REMOVE_SINCE_VERSION 2
-#endif
-
-static inline void wl_fixes_set_user_data(struct wl_fixes* wl_fixes,
-                                          void* user_data) {
-  wl_proxy_set_user_data((struct wl_proxy*)wl_fixes, user_data);
-}
-
-static inline void* wl_fixes_get_user_data(struct wl_fixes* wl_fixes) {
-  return wl_proxy_get_user_data((struct wl_proxy*)wl_fixes);
-}
-
-static inline uint32_t wl_fixes_get_version(struct wl_fixes* wl_fixes) {
-  return wl_proxy_get_version((struct wl_proxy*)wl_fixes);
-}
-
-static inline void wl_fixes_destroy(struct wl_fixes* wl_fixes) {
-  wl_proxy_marshal_flags((struct wl_proxy*)wl_fixes, WL_FIXES_DESTROY, NULL,
-                         wl_proxy_get_version((struct wl_proxy*)wl_fixes),
-                         WL_MARSHAL_FLAG_DESTROY);
-}
-
-/**
- * @ingroup iface_wl_fixes
- *
- * This request destroys a wl_registry object.
- *
- * The client should no longer use the wl_registry after making this
- * request.
- *
- * The compositor will emit a wl_display.delete_id event with the object ID
- * of the registry and will no longer emit any events on the registry. The
- * client should re-use the object ID once it receives the
- * wl_display.delete_id event.
- */
-static inline void wl_fixes_destroy_registry(struct wl_fixes* wl_fixes,
-                                             struct wl_registry* registry) {
-  wl_proxy_marshal_flags((struct wl_proxy*)wl_fixes, WL_FIXES_DESTROY_REGISTRY,
-                         NULL, wl_proxy_get_version((struct wl_proxy*)wl_fixes),
-                         0, registry);
-}
-
-/**
- * @ingroup iface_wl_fixes
- *
- * Acknowledge the removal of the specified global.
- *
- * If no global with the specified name exists or the global is not removed,
- * the wl_fixes.invalid_ack_remove protocol error will be posted.
- *
- * Due to the Wayland protocol being asynchronous, the wl_global objects
- * cannot be destroyed immediately. For example, if a wl_global is removed
- * and a client attempts to bind that global around same time, it can
- * result in a protocol error due to an unknown global name in the bind
- * request.
- *
- * In order to avoid crashing clients, the compositor should remove the
- * wl_global once it is guaranteed that no more bind requests will come.
- *
- * The wl_fixes.ack_global_remove() request is used to signal to the
- * compositor that the client will not bind the given global anymore. After
- * all clients acknowledge the removal of the global, the compositor can
- * safely destroy it.
- *
- * The client must call the wl_fixes.ack_global_remove() request in
- * response to a wl_registry.global_remove() event even if it did not bind
- * the corresponding global.
- */
-static inline void wl_fixes_ack_global_remove(struct wl_fixes* wl_fixes,
-                                              struct wl_registry* registry,
-                                              uint32_t name) {
-  wl_proxy_marshal_flags((struct wl_proxy*)wl_fixes, WL_FIXES_ACK_GLOBAL_REMOVE,
-                         NULL, wl_proxy_get_version((struct wl_proxy*)wl_fixes),
-                         0, registry, name);
-}
-
 #ifdef __cplusplus
 }
 #endif
